admin_guide/backup_restore.adoc

```diff
@@ -0,0 +1,274 @@
+
+----
+y	BuildConfig
+y	Build
+no-?	componentstatuses (aka 'cs')
+no-4	configmaps
+no-?	daemonsets (aka 'ds')
+y	DeploymentConfig
+no-?	deployments
+no-4	events (aka 'ev')
+no-4	endpoints (aka 'ep')
+no-2	horizontalpodautoscalers (aka 'hpa')
+no-1	imagestreamimages (aka 'isimage')
+y	ImageStream
+y	ImageStreamTag
+no-?	ingress (aka 'ing')
+no-2	groups
+no-?	jobs
+no-2	limitranges (aka 'limits')
+no-?	nodes (aka 'no')
+no-1	namespaces (aka 'ns')
+y	Pod
+no-?	persistentvolumes (aka 'pv')
+no-3	persistentvolumeclaims (aka 'pvc')
+no-2	policies
+no-1	projects
+no-2	quota
+no-2	resourcequotas (aka 'quota')
+no-?	replicasets (aka 'rs')
+y	ReplicationController
+no-2	rolebindings
+y	Route
+no-3	secrets
+no-2	serviceaccounts
+y	Service
+no-2	users
+----
+
+*NB: WIP*
+
+The above list is made from playing w/ the docs' team OSE 3.2 instance.
+We still need to further rationalize (and reconcile) it w/
+link:https://github.com/kubernetes/kubernetes/pull/28955#issuecomment-232737113[this comment].
+====
+
+[[cluster-restore]]
+== Cluster Restore
+
+//??? (for this section) Is the ordering (API objects, then etcd) correct?
+
+. Reinstall {product-title}.
+//??? Is there a better way to "zero out" the cluster?
+This should be done in the
+link:../install_config/install/index.html[same way]
+that {product-title} was previously installed.
+
+. Restore the certificates and keys, on each master:
++
+----
+# cd /etc/origin/master
+# tar xvf /tmp/certs-and-keys-$(hostname).tar
+----
+
+. Restore from the *etcd* backup:
++
+----
+# mv $ETCD_DATA_DIR $ETCD_DATA_DIR.orig
+# cp -Rp $ETCD_DATA_DIR.bak $ETCD_DATA_DIR
+# chcon -R --reference $ETCD_DATA_DIR.orig $ETCD_DATA_DIR
+# chown -R etcd:etcd $ETCD_DATA_DIR
+----
+// etcd 3.x will support:
+// # etcdctl restore \
+//     --backup-dir $ETCD_DATA_DIR.bak \
+//     --data-dir $ETCD_DATA_DIR
+// See also: <https://lwn.net/Articles/631630/>
+
+. Create the API objects for the cluster:
++
+----
+$ oc create -f mycluster.template.yaml
+----
+//??? Other flags?
+
+
+// ---------------------------------------------------------------------
+//??? Does the cluster need to be "quiescent" for backup/restore/both?
+//??? Generally, what are the required conditions for a successful backup/restore?
+//??? Are there other considerations for special configurations?
+//??? (meta) Is this documentation on the right track?
+
+
+[[project-backup]]
+== Project Backup
+
+A future release of {product-title} will feature specific
+support for per-project backup and restore.
+
+For now, to back up API objects at the project level,
+use `oc export` for each object to be saved.
+For example, to save the deployment configuration `frontend` in YAML format:
+
+----
+$ oc export dc frontend -o yaml > dc-frontend.yaml
+----
+
+//??? Scare quotes in next sentence because annotations are not included.
+//    For this reason, i don't want to include it...
+//
+// To back up "all" of the project:
+//
+// ----
+// $ oc export all -o yaml > project.yaml
```

Comment:
This does not backup cluster objects like namespaces, projects, and other cluster objects.